### ISSUE TITLE
columns missing after Join by same columns using mapping

### ISSUE THREAD

**Issue by antonio-yu** [pwwang/datar#122]
When tryting to join 2 different dataframe, if the columns have the same name, the they miss in the new dataframe.
Fo example, the column 'hp' will miss.
```

(mtcars
 >> select(f.cyl,f.hp) >> rename(cyl_new=f.cyl)  
>> left_join(mtcars,by={'cyl_new':'cyl','hp':'hp'})
)

```
If I only use the 'hp' to join. when by=['hp'], the column 'hp' still exists.  If by={'hp':'hp'}, it misses.

```
(mtcars
 >> select(f.cyl,f.hp) >> rename(cyl_new=f.cyl)  
>> left_join(mtcars,by={'hp':'hp'})
)

```

```
(mtcars
 >> select(f.cyl,f.hp) >> rename(cyl_new=f.cyl)  
>> left_join(mtcars,by=['hp'])
)
```

**Comment by pwwang** [comment]
Very nice catch!

This will be fixed at v0.8.5. 

A workaround for now is to use list: `['x', 'y']`, instead of a dict: `{'x': 'x', 'y': 'y'}`

**Comment by pwwang** [comment]
Fixed at [v0.8.5](#125)
